### PR TITLE
Fleet: Add actions timeout mapping

### PR DESCRIPTION
**Pull request by aleksmaus**
## What does this PR do?

Add the new "timeout" field to the ".fleet-actions" index mapping.

## Related issues

- Relates https://github.com/elastic/kibana/issues/105477

## Screenshots

Verified the action with the integer timeout value is saved 
Screen Shot 2021-07-22 at 6 55 38 AM

Verified the .fleet-actions index mapping reflects the change
Screen Shot 2021-07-22 at 6 54 58 AM

### CODE REVIEW

**Comment by gwbrown** [comment]
@aleksmaus As Jay Modi has recent switched teams, could you ping me or @williamrandolph on future reviews?

**Comment by gwbrown** [comment]
@elasticmachine update branch

**Comment by aleksmaus** [comment]
@elasticmachine update branch

**Comment by williamrandolph** [comment]
@elasticmachine update branch

**Comment by aleksmaus** [comment]
@elasticmachine update branch

File: x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/Fleet.java [hunk 1]
```diff
@@ -73,7 +73,7 @@
  */
 public class Fleet extends Plugin implements SystemIndexPlugin {
 
-    private static final int CURRENT_INDEX_VERSION = 7;
+    private static final int CURRENT_INDEX_VERSION = 8;
```
Comment (williamrandolph) [inline x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/Fleet.java:76]: Changing this value is causing test failures, because we have the `-7`suffix hardcoded in some of them. The current index version is used in the index name, and I think we want to wait to bump it to 8 until we are reindexing for the 8.x release.

Since this mapping change is only adding a value and not changing a type, the `SystemIndexManager` class should cleanly apply the change. It will use the Elasticsearch version number under the hood to determine whether the mappings are out of date, and won't need to change the index name.

In short, I think we can back out the change in this line and get tests to pass.

**Review by williamrandolph — CHANGES_REQUESTED** [review]
I apologize very much for the delayed review. I don't think we need to increment the index version here, but the change itself should be fine.

File: x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/Fleet.java (hunk 1 above)
Comment (aleksmaus) [reply x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/Fleet.java]: Thank you for the feedback. Rolled back the version bump.

**Review by williamrandolph — APPROVED** [review]
Thanks for the change! I think this is ready to merge. I'm going to tag this for automatic backport, and I'm happy to handle the merge if you'd like me to.
